test/runtests.jl

```diff
@@ -57,6 +57,29 @@ function test_explicit_ordering(T::Type = UInt)
             xs[1] = x1
             @test rmw(ptr, x2, acquire) === x1
             @test xs[1] === op(x1, x2)
+
+            # Test syncscopes.
+            if (op == +) || (op == -)
+                xs[1] = x1
+                @test UnsafeAtomics.modify!(
+                    ptr,
+                    op,
+                    x2,
+                    seq_cst,
+                    UnsafeAtomicsLLVM.Internal.SyncscopeSystem,
```

Comment:
The `Internal` leaking into user code doesn't seem ideal. Wouldn't these have to be part of the public (if unexported) API?

Reply:
Would using raw `Val{...}` be fine?

Reply:
Yeah, that's probably fine.